### PR TITLE
fix: add deprecated warning message in insights.combiners.mounts

### PR DESCRIPTION
**Pull request by chenlizhong**
Signed-off-by: Chen lizhong <lichen@redhat.com>

### All Pull Requests:

Check all that apply:

* [ ] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Before remove the combiner.mounts, give deprecate warning by this MR.
https://github.com/RedHatInsights/insights-core/pull/3606

### CODE REVIEW

File: insights/combiners/mounts.py [hunk 1]
```diff
@@ -42,6 +43,7 @@ class Mounts(object):
     """
 
     def __init__(self, binmount, procmounts, mountinfo):
+        deprecated(Mounts, "Use the parsers in insights.mount module instead", "3.1.0")
```
Comment (psachin) [inline insights/combiners/mounts.py:46]: @chenlizhong Please make it `3.1.25` which means we can remove it after the release of `3.2.0` which is roughly six months from now.

Comment (chenlizhong) [reply insights/combiners/mounts.py]: Updated

**Review by xiangce — CHANGES_REQUESTED** [review]
It's also recommended to add related `warning`s to the docstring of the combiner when deprecating it.

**Review by xiangce — APPROVED** [review]
Good to me. But TBH, a `Wanning` would be better than a `Note` in the doc.

**Pull request merged**